proto/spire/plugin/server/credentialcomposer/v1/credentialcomposer.proto

```diff
@@ -0,0 +1,179 @@
+    // The attributes for the X509-SVID. If not included in the response, the
+    // default attributes sent in the request will be used.
+    X509SVIDAttributes attributes = 1;
+}
+
+message ComposeWorkloadJWTSVIDRequest {
+    // The default attributes for the JWT-SVID.
+    JWTSVIDAttributes attributes = 1;
+
+    // The SPIFFE ID of the workload.
+    string spiffe_id = 2;
+}
+
+message ComposeWorkloadJWTSVIDResponse {
+    // The attributes for the JWT-SVID. If not included in the response, the
+    // default attributes sent in the request will be used.
+    JWTSVIDAttributes attributes = 1;
+}
+
+message X509CAAttributes {
+    // The subject of the X509 CA.
+    DistinguishedName subject = 1;
+
+    // Zero or more policy identifiers (OIDs) to apply to the CA.
+    repeated string policy_identifiers = 2;
+
+    // Zero or more extensions to apply to the X509 CA. These will override
+    // any extensions otherwise added by the other fields.
+    repeated X509Extension extra_extensions = 3;
+}
+
+message X509SVIDAttributes {
+    // The subject of the X509-SVID.
+    DistinguishedName subject = 1;
+
+    // Zero or more DNS SANs to apply to the X509-SVID.
+    repeated string dns_sans = 2;
+
+    // Zero or more extensions to apply to the X509-SVID . These will override
+    // any extensions otherwise added by the other fields.
+    repeated X509Extension extra_extensions = 3;
+}
+
+message X509Extension {
+    // The OID of the X.509 extension (e.g. "1.2.3.4")
+    string oid = 1;
+
+    // Opaque value of the extension. No validity checking is performed on
+    // this value. Plugin implementors must ensure they are providing well
+    // formed values for the given extension OID.
+    bytes value = 2;
+
+    // Whether or not the extension is critical, i.e., must be
+    // handled/understood by verifiers or not.
+    bool critical = 3;
+}
+
+message DistinguishedName {
+    // Zero or more country designations.
+    repeated string country = 1;
+
+    // Zero or more organization designations.
+    repeated string organization = 2;
+
+    // Zero or more organizational unit designations.
+    repeated string organizational_unit = 3;
+
+    // Zero or more locality designations.
+    repeated string locality = 4;
+
+    // Zero or more province designations.
+    repeated string province = 5;
+
+    // The serial number designation. The attribute is only set if this field
+    // is non-empty.
+    string serial_number = 8;
+
+    // The common name designation. The attribute is only set if this field is
+    // non-empty.
+    string common_name = 9;
+
+    // Extra names, determined by oid and value, to be added to the
+    // distinguished names. This field is to support names not covered by the
+    // DistinguishedName message. It will override values specified in other
+    // fields in the DistinguishedName if the attributes overlap.
+    repeated AttributeTypeAndValue extra_names = 10;
```

Comment:
it is possible we introduce a feature where we start to use `UniqueIdentifier`, may we add it here too?

Reply:
If we introduce the unique_id, I'd probably think about adding it to DN as a first-class field instead of putting it inside extra_names.

Reply:
yeap that was I was thinking too